spyder/widgets/editor.py

```diff
@@ -1583,6 +1591,12 @@ def save(self, index=None, force=False):
             return self.save_as(index=index)
         if self.always_remove_trailing_spaces:
             self.remove_trailing_spaces(index)
+        if self.convert_eol_on_save:
+            # hack to account for the fact that the config file saves

```

Comment:
Maybe this logic could be moved to a function in `spyder/utils/sorcecode.py`, `EOL_CHARS` could be changed, to also include `CR/LF/CRLF`:

`EOL_CHARS = (("\r\n", 'nt', 'CRLF'), ("\n", 'posix', 'LR'), ("\r", 'mac', 'CR'))`